### PR TITLE
Feature: improve branch coverage

### LINKED ISSUE #6020: Feature: improve branch coverage
We want to improve branch coverage, starting with the folder data_structures/queue. 
The goal is to write proper tests and documentation as per the contribution guidelines.

For example, inside the folder data_structures/queue, while `circular_queue.py`, `circular_queue_linked_list.py`, `double_ended_queue.py` and `linked_queue.py` pass the tests, there are problems in the other programs.

- normalize all tests with doctests (no print statement as it can be the case today)
- increase branch coverage where it is possible
- fix failing tests where we find them.

Our contribution:
- data_structures/hashing (entire repository)
- data_structures/linked_list  (entire repository)
- strings (min_cost_conversion.py, knutt_morris_pratt.py)
- data_structures/binary_tree/number_of_possible_binary_trees.py
- matrix/count_islands_in_matrix.py
- matrix/matrix_class.py
- divide_and_conquer/convex_hull.py
- divide_and_conquer/closest_pair_of_points.py
- data_structures/binary_tree/red_black_tree.py
- data_structures/binary_tree/treap.py
- data_structures/queue/priority_queue_using_list.py 
- data_structures/queue/queue_on_pseudo_stack.py 
- data_structures/binary_tree/binary_search_tree.py

### PR DESCRIPTION
**Pull request by Dajamante**
### Describe your change:

closes #6020
Please see issue: 

* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/binary_tree/red_black_tree.py [hunk 1]
```diff
@@ -93,6 +93,16 @@ def insert(self, label: int) -> RedBlackTree:
         rotations necessary to maintain balance, and then returns the
         new root to this subtree (likely self).
         This is guaranteed to run in O(log(n)) time.
+
```
Comment (poyea) [inline data_structures/binary_tree/red_black_tree.py:96]:
```suggestion
```

File: data_structures/queue/priority_queue_using_list.py [hunk 2]
```diff
@@ -141,9 +156,16 @@ class ElementPriorityQueue:
...
+    priority_queue_using_list.UnderFlowError: The queue is empty
     >>> print(epq)
     []
+
+
+    >>> epq2 = ElementPriorityQueue()
+    >>> q = [[epq2.enqueue(11)] for _ in range(101)]
+    Traceback (most recent call last):
+    ...
+    priority_queue_using_list.OverFlowError: Maximum queue size is 100
```
Comment (poyea) [inline data_structures/queue/priority_queue_using_list.py:168]: `FAILED data_structures/queue/priority_queue_using_list.py::data_structures.queue.priority_queue_using_list.ElementPriorityQueue`

File: data_structures/queue/priority_queue_using_list.py [hunk 3]
```diff
@@ -78,6 +78,21 @@ def enqueue(self, priority: int, data: int) -> None:
...
+        >>> fpq.enqueue(3, 10)
+        Traceback (most recent call last):
+        ...
+        ValueError: Valid priorities are 0, 1, and 2
+        >>> fpq = FixedPriorityQueue()
+        >>> q = [fpq.enqueue(0,5) for _ in range(101)]
+        Traceback (most recent call last):
+        ...
+        OverflowError: Maximum queue size is 100
+
```
Comment (poyea) [inline data_structures/queue/priority_queue_using_list.py:95]: `FAILED data_structures/queue/priority_queue_using_list.py::data_structures.queue.priority_queue_using_list.FixedPriorityQueue`

**Comment by poyea** [comment]
Please fix the build checks.

File: data_structures/binary_tree/treap.py [hunk 4]
```diff
@@ -1,12 +1,30 @@
 from __future__ import annotations
 
+import re
 from random import random
 
+__all__ = ("re",)
```
Comment (cclauss) [inline data_structures/binary_tree/treap.py:6]: Why do we need this??

File: data_structures/hashing/double_hash.py [hunk 5]
```diff
@@ -11,26 +11,79 @@ class DoubleHash(HashTable):
...
-    def __hash_function_2(self, value, data):
-
+    def _hash_function_2(self, value, data):
+        """
+        >>> hashTable = DoubleHash(3)
+        >>> hashTable._hash_function_2(1, 4) # next_prime_gt = 3, return 3 - (4 % 3) = 2
+        2
+        >>> hashTable = DoubleHash(5)
+        >>> hashTable._hash_function_2(3, 8) # next_prime(3) = 3, return 3 - (8 % 3) = 1
+        1
```
Comment (cclauss) [inline data_structures/hashing/double_hash.py:21]:
```suggestion
        >>> hashTable._hash_function_2(1, 4)  # next_prime_gt = 3, return 3 - (4 % 3) = 2
        2
        >>> hashTable = DoubleHash(5)
        >>> hashTable._hash_function_2(3, 8)  # next_prime(3) = 3, return 3 - (8 % 3) = 1
        1
```

**Review by cclauss — APPROVED** [review]

**Comment by cclauss** [comment]
All checks fail?

**Comment by SnowdenSa** [comment]
Nice work.

**Comment by poyea** [comment]
Let's fix the tests? - this would be great to have

**Comment by Dajamante** [comment]
I agree! I will not have time unfortunately :(. But there are two tests that are failing according to the report.
================ 2 failed, 1117 passed, 108 warnings in 32.80s =================

**Comment by poyea** [comment]
Yes, but maintainers do not have the edit permission, so if you could please fix those 2 issues

**Comment by cclauss** [comment]
In preparation for that denial-of-service attack that is [Hacktoberfest](https://hacktoberfest.com), I am closing all pull requests that have failing tests.

**Pull request closed without merging**